### ISSUE TITLE
Request gets 404 error

### ISSUE THREAD

**Issue by crayne** [ResearchStack/SampleApp#3]
````
<-- 404 Not Found https://webservices-staging.sagebridge.org/v3/auth/signUp (851ms)
````
POST:
````
--> POST https://webservices-staging.sagebridge.org/v3/auth/signUp HTTP/1.1
Content-Type: application/json
Content-Length: 91
User-Agent: STUDY_NAME_HERE/1
Bridge-Session: 
{"email":"susancrayne@yahoo.com","password":"1234","study":"STUDY_ID_HERE","type":"SignUp"}
--> END POST (91-byte body)
````

**Comment by wdziemia** [comment]
Please see instructions within this file https://github.com/ResearchStack/SampleApp/blob/0c19a34f5a0b4bcee4ea824a139ba5446b6f5b34/app/gradle.properties.sample

**Comment by crayne** [comment]
I do have a gradle.properties file -- I put dummy strings in it
If the studyId and studyName need to be real, how do I find out what they should be?

**Comment by crayne** [comment]
I also tried:
studyId="STUDY_ID_HERE"
studyName="STUDY_NAME_HERE"

**Comment by mesa72** [comment]
I have experienced the same issue. 
POST https://webservices-staging.sagebridge.org/v3/auth/signUp HTTP/1.1
Content-Type: application/json
Content-Length: 97
User-Agent: STUDY_NAME_HERE/1
Bridge-Session: 
{"email":"matthew.mclaughlin@cshs.org","password":"1111","study":"STUDY_ID_HERE","type":"SignUp"}
END POST (97-byte body)
404 Not Found https://webservices-staging.sagebridge.org/v3/auth/signUp
Server: Cowboy

Thank you,

**Comment by wdziemia** [comment]
The following values will **not** work. 

```
studyId="STUDY_ID_HERE"
studyName="STUDY_NAME_HERE"
```

**You must change these to what is found within the [Bridge Study Manager](**research.sagebridge.org**)**. The studyName can be found under Study > Settings > General > "Study name". The study identifier is also on the same page and is shown in the blurb on the right side of the page.

**Comment by crayne** [comment]
research.sagebridge.org requires that one log in.  Same problem -- no password.  Previous versions of the sample app did not have this problem.  I assume that the bridge communication piece was just implemented.  Can that implementation be removed, or can you tell us how to change the code ourselves?  I have looked into this, and it is not obvious.
Taking it back a step, inquiring minds would like to know, what is the "bridge" and to what extent is it coupled to ResearchStack?

**Comment by wdziemia** [comment]
Correct, we were using the credentials for mole mapper while SampleApp was in development. This needed to be removed for when we open sourced the code. Sage Bridge is the backend servcie that can be used for sign-in, sign-up, consent, and data upload. It only exists within the App and you are free to implement your own code for data upload. While it can be removed,  it will be quite a bit of work. You will need to implement DataProvider and return that within the ResearchStack instance.

**Comment by crayne** [comment]
Thank you.  It seems to me that SampleApp now has substantially reduced utility for new  app developers.

**Comment by mikecarroll** [comment]
Yes, this is a limitation with the SampleApp, as it stands. We built our tests apps against Sage Bridge (which MoleMapper uses) but haven't had the cycles to build a version that doesn't require a Sage account or backend for the signup process yet.

Two thoughts:

(1) ResearchStack is backend agnostic so have a mock DataProvider for the SampleApp and basic dev would be an amazing conribution, if somebody wants to build it. @dariosalviwork [originally proposed the idea here](https://github.com/ResearchStack/ResearchStack/issues/56), and we would be excited to accept a PR to the SampleApp for it.

(2) Since @crayne raised this issue, I've been talking with Sage about making the initial dev process against Bridge easier and they have some potential solutions in the works that I hope we can implement here soon.

**Comment by DP0326** [comment]
can anyone repeat the solution for this problem?? I am not being able to solve this problem.

**Comment by DP0326** [comment]
I have created a file named gradle.properties and have included the following values: 

studyId="bridge_study_id"
studyName="bridge_study_name"

but this doesn't seem to solve the problem. Can you please give me any suggestion to solve the problem? @wdziemia

**Comment by liujoshua** [comment]
@DP0326 I'd be happy to help but won't have time until late next week at the earliest.

For now, I'd suggest looking at the pull requests [for this repo](https://github.com/ResearchStack/MoleMapperAndroid/pulls). This should address most of the issues.

**Comment by DP0326** [comment]
Hi,

Thank you for your prompt reply. But I need to solve this problem by this
Friday. Anykind of help will be appreciated.
